pkg/apply/taskrunner/task.go

```diff
@@ -111,16 +111,23 @@ func (w *WaitTask) setTimer(taskContext *TaskContext) {
 		// Timeout is cancelled.
 		<-timer.C
 		select {
-		// We only send the taskResult if no one has gotten
+		// We only send the TimeoutError to the eventChannel if no one has gotten
 		// to the token first.
 		case <-w.token:
-			taskContext.TaskChannel() <- TaskResult{
-				Err: &TimeoutError{
-					Identifiers: w.Ids,
-					Timeout:     w.Timeout,
-					Condition:   w.Condition,
+			err := &TimeoutError{
```

Comment:
Nit: I like that we no longer need to amend certain error types in the runner itself. In this case, it seems like we can just create the `TimeoutError` directly without having to have the two steps here.

Reply:
`amendTimeoutError` should probably be refactored to just have a function that returns the IDs that timed out, like `pending` except also strip out the ones that are Current or NotFound (for apply or prune).